mdata/importer/converter.go

```diff
@@ -132,6 +122,15 @@ func (c *converter) findSmallestLargestArchive(spp, nop uint32) (int, int) {
 		smallestArchiveIdx = i
 	}
 
+	// find smallest archive that still contains enough data to satisfy requested range
```

Comment:
if smallest already points to an archive that has enough TTL, then that's the only thing we need. we don't need any other archives

Reply:
yep, that's why now it's starting to check from smallest and if it has enough TTL then it uses smallest as largest as well